extensions/amp-recaptcha-input/0.1/amp-recaptcha-service.js

```diff
@@ -13,3 +13,131 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Service for recaptcha components
+ * interacting with the 3p recaptcha bootstrap iframe
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {dev} from '../../../src/log';
+import {getIframe} from '../../../src/3p-frame';
+import {getService, registerServiceBuilder} from '../../../src/service';
+import {listenFor} from '../../../src/iframe-helper';
+import {loadPromise} from '../../../src/event-helper';
+import {removeElement} from '../../../src/dom';
+
+export class AmpRecaptchaService {
+
+  /**
+   * @param {!Window} window
+   */
+  constructor(window) {
+
+    /** @const @private {!Window} */
+    this.win_ = window;
+
+    /** @const @private {!Element} */
+    this.body_ = this.win_.document.body;
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframeLoadPromise_ = null;
+
+    /** @private {number} */
+    this.registeredElementCount_ = 0;
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+
+    /** @private {!Deferred} */
+    this.willBeReady_ = new Deferred();
+  }
+
+  /**
+   * Function to register as a dependant of the AmpRecaptcha serivce.
+   * Used to create/destroy recaptcha boostrap iframe.
+   * @param {!Element} element
+   * @return {Promise}
+   */
+  register(element) {
+    this.registeredElementCount_++;
+    if (!this.iframe_) {
```

Comment:
nit: It would be easier to understand the code with logic like
```
if (!this.iframeLoadPromise_) {
 // initilize
}
return this.iframeLoadPromise_
```
given that `this.iframe_` and `this.iframeLoadPromise_` will always be created and destroyed at the same time.